pkg/cryptoauthlib/Kconfig

```
@@ -0,0 +1,40 @@
+# Copyright (c) 2022 Inria
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+config PACKAGE_CRYPTOAUTHLIB
+    bool "Microchip CryptoAuthentication Library package"
+    depends on TEST_KCONFIG
+    select MODULE_AUTO_INIT_SECURITY
+    select MODULE_CRYPTOAUTHLIB_CONTRIB
+
+config MODULE_CRYPTOAUTHLIB_CONTRIB
+    bool
+    depends on TEST_KCONFIG
+    select MODULE_ZTIMER
+    select MODULE_ZTIMER_USEC
+    select MODULE_ZTIMER_PERIPH_TIMER
+    select MODULE_PERIPH_I2C
+    select MODULE_PERIPH_I2C_RECONFIGURE if HAS_PERIPH_I2C_RECONFIGURE
+
+config MODULE_CRYPTOAUTHLIB_TEST
+    bool "Module for cryptoauthlib tests"
+    depends on TEST_KCONFIG
+    select MODULE_CRYPTOAUTHLIB_TEST_JWT
+    select MODULE_CRYPTOAUTHLIB_TEST_TNG
+    select MODULE_CRYPTOAUTHLIB_TEST_ATCACERT
```

```suggestion
menuconfig PACKAGE_CRYPTOAUTHLIB
    bool "Microchip CryptoAuthentication Library package"
    depends on TEST_KCONFIG
    select MODULE_AUTO_INIT_SECURITY
    select MODULE_CRYPTOAUTHLIB_CONTRIB

if PACKAGE_CRYPTOAUTHLIB

config MODULE_CRYPTOAUTHLIB_TEST
    bool "Module for cryptoauthlib tests"
    depends on TEST_KCONFIG
    select MODULE_CRYPTOAUTHLIB_TEST_JWT
    select MODULE_CRYPTOAUTHLIB_TEST_TNG
    select MODULE_CRYPTOAUTHLIB_TEST_ATCACERT
    
endif # PACKAGE_CRYPTOAUTHLIB

config MODULE_CRYPTOAUTHLIB_CONTRIB
    bool
    depends on TEST_KCONFIG
    select MODULE_ZTIMER
    select MODULE_ZTIMER_USEC
    select MODULE_ZTIMER_PERIPH_TIMER
    select MODULE_PERIPH_I2C
    select MODULE_PERIPH_I2C_RECONFIGURE if HAS_PERIPH_I2C_RECONFIGURE
```
If you do it like this, the test option is hidden from the menu until you choose the package.

---

I pushed an alternative without the menu (because I don't see the point in have a menu for configuring modules).

---

It is more intuitive to have modules that depend on the package inside the `menuconfig`. That is the way we have been doing it so far, instead of having a huge flat list that keeps changing every time you enable/disable a module. It does not affect `.config` users, and helps `menuconfig` users.